### PR TITLE
config: add support for StopSignal

### PR DESCRIPTION
**Pull request by q384566678**
Add a stop-singnal option for buildah-config.

Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

File: docs/buildah-config.md [hunk 1]
```diff
@@ -72,6 +72,10 @@ The shell instruction allows the default shell used for the shell form of comman
 
 Note: this setting is not present in the OCIv1 image format, so it is discarded when writing images using OCIv1 formats.
 
+**--stop-signal** *signal*
+
+Specifies a *signal* and the system call this *signal* that will be sent to the container to exit.
```
Comment (TomSweeneyRedHat) [inline docs/buildah-config.md]: Suggest: "Sets the system call *signal* that will be sent to the container to exit."

**Comment by TomSweeneyRedHat** [comment]
Ditto @ripcurld's thought on test.  The test failures don't seem to be related to this PR, looks like some kind of unrelated issue on the test systems.  @q384566678 thanks for the PR!

**Comment by rhatdan** [comment]
Lets get this in ASAP,  Now that we have Multi-stage builds we want to cut a new release and this is something we should support.

File: cmd/buildah/config.go [hunk 2]
```diff
@@ -63,6 +63,10 @@ var (
 			Name:  "shell",
 			Usage: "add `shell` to run in containers",
 		},
+		cli.StringFlag{
+			Name:  "stop-signal",
+			Usage: "add `signal` to stop containers",
```
Comment (rhatdan) [inline cmd/buildah/config.go]: set default stop signal for container.  This signal will be sent when container is stopped, default is SIGINT.

File: docs/buildah-config.md (hunk 1 above)
Comment (rhatdan) [reply docs/buildah-config.md]: I would explain it further as I did above.



set default stop signal for container. This signal will be sent when container is stopped, default is SIGINT.

**Comment by rhatdan** [comment]
A good test would be to just check buildah inspect to make sure the stop signal is set.

**Comment by TomSweeneyRedHat** [comment]
Test issues should be fixed with #572

**Comment by rhatdan** [comment]
@q384566678 Could you update your PR to pull in the fix for the docker change.

**Comment by q384566678** [comment]
@rhatdan What's the change for docker?

**Comment by q384566678** [comment]
I updated the explanation and added related tests.

**Comment by rhatdan** [comment]
@q384566678 I think you figured it out, but docker no longer wants you to do `--password foobar` on the command line.

LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**